packages/core/core-flows/src/fulfillment/steps/validate-shipping-option-prices.ts

```diff
@@ -0,0 +1,53 @@
+import { FulfillmentWorkflow } from "@medusajs/framework/types"
+import { MedusaError, Modules } from "@medusajs/framework/utils"
+import { StepResponse, createStep } from "@medusajs/framework/workflows-sdk"
+
+export const validateShippingOptionPricesStepId =
+  "validate-shipping-option-prices"
+
+/**
+ * Validate that regions exist for the shipping option prices.
+ */
+export const validateShippingOptionPricesStep = createStep(
+  validateShippingOptionPricesStepId,
+  async (
+    options: {
+      prices?: FulfillmentWorkflow.UpdateShippingOptionsWorkflowInput["prices"]
+    }[],
+    { container }
+  ) => {
+    const allPrices = options.flatMap((option) => {
+      if (!option.prices) {
+        return []
+      }
+      return option.prices
+    })
+
+    const regionIdSet = new Set<string>()
+
+    allPrices.forEach((p) => {
+      if ("region_id" in p && p.region_id) {
+        regionIdSet.add(p.region_id)
+      }
+    })
+
+    const regionService = container.resolve(Modules.REGION)
+    const regionList = await regionService.listRegions({
```

Comment:
Sug: we should probably check the length before calling the service just as a best practice, i think we are handling it under the hood but if a module is replaced it might not always be the case wdyt? Also, it removes a call to the service and we can return early if there is nothing to check